lib/api/api.js

```diff
@@ -184,14 +186,103 @@ const api = {
             }
             return { returnTagCount, isImplicitDeny };
         }
+        let formDataParser;
+        let fileEventData = null;
+
+        if (apiMethod === 'objectPost') {
+            formDataParser = busboy({ headers: request.headers });
+        }
 
         return async.waterfall([
+            next => {
+                if (apiMethod === 'objectPut' || apiMethod === 'objectPutPart') {
+                    return next(null);
+                }
+                if (apiMethod === 'objectPost' && request.headers['content-type'].includes('multipart/form-data')) {
+                    writeContinue(request, response);
+
+                    request.formData = {};
+                    let totalFieldSize = 0;
+                    const MAX_FIELD_SIZE = 20 * 1024; // 20KB
+                    formDataParser.on('field', (fieldname, val) => {
```

Comment:
< [!CAUTION]
> The aws doc says there are 2 mandatory fields, and some conditional: https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPOST.html#RESTObjectPOST-requests-form-fields
> 
> There is no validation right now
>
> A request without `key` field should return a 400 `InvalidArgument` but right now it accepts the request and save the object with a key `null`

Reply:
57f34b9403389aba4718887f90896dfb7d5c69b6

Reply:
This is still not ignoring other fields

Reply:
You should use the variable `fieldname` provided in both events `field` and `file` to accept only the recognized field defined in AWS documentation (for now you only handle `key`, and `file` fieldnames) and validation should better be performed outside any of those events as values can be passed either as text string or file

Reply:
Unrecognized fields simply will not be used/stored anywhere; it might even be less optimal to check them given it means making a check against recognized fields for every field we receive...
Current behaviour in effect is that unrecognized fields are ignored, unless I'm missing something.

In doubt, I'm writing up a solution, but I fail to see a scenario where this is the case?

> values can be passed either as text string or file

The parser makes checks based on the content header of each form-data boundary. Is it possible for fields keys to be missed because of their format? I'm not sure I follow

Reply:
I'll try to describe more in another comment

Reply:
see https://github.com/scality/cloudserver/pull/5601#discussion_r1690995359 I'm closing this